src/base/net/private/geoipdatabase.h

```diff
@@ -29,13 +29,13 @@
 #ifndef GEOIPDATABASE_H
 #define GEOIPDATABASE_H
 
-#include <QtGlobal>
 #include <QCoreApplication>
+#include <QtGlobal>
```

Comment:
Is it really needed here?

Reply:
Compiles without it so i guess not.

Reply:
But this can't be the criteria for it's removal right? Because it compiles even if i remove it from many other files as well.

Reply:
And while we're on the subject, i re-read the guidelines and i saw this
[One exception is the header containing the library version (for example, QtGlobal), this particular header isn't constrained by the aforementioned order.](https://github.com/qbittorrent/qBittorrent/blob/master/CODING_GUIDELINES.md#7-header-inclusion-order)
Does that mean it can go anywhere and i shouldn't bother with its position?

Reply:
>But this can't be the criteria for it's removal right? Because it compiles even if i remove it from many other files as well.

Actually, we need include QtGlobal only in one case, when we have to use some global Qt type/typedef (e.g. quint32) but we don't include any other Qt header.

Reply:
>Does that mean it can go anywhere and i shouldn't bother with its position?

It means that if some other includes depend on Qt version we can put QtGlobal first (out of standard order).

Reply:
Ok thanks for the explanation. I however, have seen you guys talk about including or not some headers that are indirectly included in some other header that's needed anyway. Doesn't this fall in that category? I mean such types (quint32) are used in this file.

Reply:
Personally I would keep the QtGlobal header.

>Since QtGlobal "is included by most other Qt header files" we don't need include it directly in most cases. 

If we do this, then we will be fragile to Qt changes, i.e. when they modify their header includes.

Reply:
>Unlike any other Qt header globality of QtGlobal is documented behavior

It only says: `It is included by most other Qt header files`. I read "most" =\\= "not all".

I don't really care if QtGlobal here is removed as long as it still compiles.
I mean, if we forgot to include it, then so be it, as long as code still compiles; if we included it and uses some of its defines directly, why not keep it as we do to other headers?

Reply:
>if we included it already and uses some of its defines directly, why not just keep it as we do to other headers?

That sounds reasonable. Agreed.

Reply:
I'll just move it around then.